Tools/CISupport/build-webkit-org/steps.py

```diff
@@ -1565,6 +1565,201 @@ def getResultSummary(self):
         return super().getResultSummary()
 
 
+class ArchiveStaticAnalyzerResults(shell.ShellCommandNewStyle):
+    command = ["Tools/Scripts/generate-static-analysis-archive", "--id-string", WithProperties("Build #%(buildnumber)s"),
+               "--output-root", "scan-build-output", "--destination", "/tmp/static-analysis.zip"]
+    name = "archive-static-analyzer-results"
+    description = ["archiving static analyzer results"]
+    descriptionDone = ["archived static analyzer results"]
+
+
+class UploadStaticAnalyzerResults(UploadTestResults):
+    name = "upload-static-analyzer-results"
+    workersrc = "/tmp/static-analysis.zip"
+
+
+class ScanBuildSmartPointer(shell.ShellCommand):
+    name = "scan-build-smart-pointer"
+    description = ["scanning with static analyzer"]
+    descriptionDone = ["scanned with static analyzer"]
+    analyzeFailed = False
+    bugs = 0
+
+    def __init__(self, **kwargs):
+        kwargs['timeout'] = 2 * 60 * 60
+        super(ScanBuildSmartPointer, self).__init__(**kwargs)
+        self.commandFailed = False
+
+    def start(self):
+        self.command = '/bin/rm -rf {output_dir}; '.\
+            format(output_dir=os.path.join(self.getProperty('builddir'), 'build/scan-build-output'))
```

Comment:
Let's try to use f-string instead of .format().
Earlier, I tried to convert all the existing .format() in this file in https://commits.webkit.org/272914@main